src/pricing.ts

```diff
@@ -476,6 +477,46 @@ const recomputeDetailTotalsFromCompositePrice = (
   }, details || initialPricingDetails);
 };
 
+const BLACK_LISTED_KEYS: Set<keyof Price> = new Set([
+  '_org',
+  '_schema',
+  '_created_at',
+  '_updated_at',
+  '_owners',
+  '_acl',
+  '_acl_sync',
+  '_viewers',
+  '_relations',
+  '$relation',
+  'file',
+  '_files',
+  'workflows',
+  '_slug',
+]);
+
+/**
+ * Maps the whitelisted keys from the _price that we want available, avoiding having unnecessary data in the metadata.
+ *
+ *? @todo We should define _price as not being an optional key in PriceItemDto since there is no case where is optional
+ *? Additionally, we should also define the _product as a separate interface to avoid `PriceItemDto['_product']` type definitions.
+ */
+export const getMappedPricing = (price: Price): Price => {
+  const mappedPricing: Price = {} as Price;
+
+  for (const key in price) {
+    if (!BLACK_LISTED_KEYS.has(key) && price[key] !== '') {
+      mappedPricing[key] = price[key];
+    }
+
+    // Checks if price has price_components and if so iterates over them to extract whitelisted keys
+    if (key === 'price_components' && price[key]) {
```

Comment:
We would be safer by ensuring it's an array 🙂 

```suggestion
    if (key === 'price_components' && price[key] && Array.isArray(price[key])) {
```

Reply:
Ah! Nice catch. Good idea, thanks you